gamemodes/terrortown/gamemode/shared/sh_inventory.lua

```diff
@@ -0,0 +1,82 @@
+local maxSpecialSlots = CreateConVar("ttt2_max_special_slots", "2", {FCVAR_NOTIFY, FCVAR_ARCHIVE}, "Maximum amount of special weapons, a player can carry (-1 = infinite)")
+local maxExtraSlots = CreateConVar("ttt2_max_extra_slots", "-1", {FCVAR_NOTIFY, FCVAR_ARCHIVE}, "Maximum amount of extra weapons, a player can carry (-1 = infinite)")
+
+local function InventorySlotFromKind(kind)
+	if kind == WEAPON_EQUIP1 or kind == WEAPON_EQUIP2 then
+		return WEAPON_SPECIAL
+	elseif kind ==  WEAPON_ROLE or kind == WEAPON_NONE then
+		return WEAPON_EXTRA
+	elseif kind > WEAPON_NONE then
+		return WEAPON_EXTRA
+	else
+		return kind
+	end	
+end
+
+--we appeareantly need to have that, because added and removed entities are sometimes broken on client side and need to get updated later
+--additionally spawn can be called when a player is alive in which case he doesn't loose weapons beforehand
+function CleanupInventoryIfDirty(ply)
+	if !ply.inventory || ply.refresh_inventory_cache then
```

Comment:
The idea behind that is a kind of caching functionality, which only updates the inventory if needed and dirty (e.g. GetInventory is called). Updating weapons work great on Server side, but doesn't on Client, because Entities are updated late/are invalid for some reason.

A solution to beautify the code would be to always use caching (not only when something is invalid). However, this would probably be less runtime efficient.